Framework/API/src/FileFinder.cpp

```diff
@@ -605,12 +605,31 @@ FileFinderImpl::findRuns(const std::string &hintstr,
       if (runEndNumber < runNumber) {
         throw std::invalid_argument("Malformed range of runs: " + *h);
       }
+      std::string previousPath, previousExt;
       for (int irun = runNumber; irun <= runEndNumber; ++irun) {
         run = std::to_string(irun);
         while (run.size() < nZero)
           run.insert(0, "0");
+
+        // Quick check if file can be created from previous successfully found
+        // path/extension
+        if (!previousPath.empty() && !previousExt.empty()) {
+          try {
+            const Poco::File file(previousPath + p1.first + run + previousExt);
+            if (file.exists()) {
+              res.emplace_back(previousPath + p1.first + run + previousExt);
+              continue;
+            }
+          } catch (...) {
+          }
```

Comment:
If we have caught an error here perhaps we should clear the previous path and extensions as they likely causes issues if the next one is okay it'll cache again anyway?